src/storage/context/StorageExpressionContext.h

```diff
@@ -316,6 +324,9 @@ class StorageExpressionContext final : public ExpressionContext {
 
   // name -> Value with multiple versions
   std::unordered_map<std::string, std::vector<Value>> valueMap_;
+
+  // Expression value map that stores the value of innerVar
+  std::unordered_map<std::string, Value> exprValueMap_;
```

Comment:
Why is it MT-safe? And I don't see you create different `ExprCtx` for each thread.

Reply:
See https://github.com/vesoft-inc/nebula/blob/f2cee664e78e1f855e3f5cec0ad60797291bdc53/src/graph/executor/query/ProjectExecutor.cpp#L28
Every thread has its own QueryExpressionContext.